examples/chip-tool-darwin/commands/clusters/ClusterCommandBridge.h

```
@@ -0,0 +1,135 @@
+        AddArgument("payload", &mPayload);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+        ModelCommand::AddArguments();
+    }
+
+    ClusterCommand(chip::ClusterId clusterId)
+        : ModelCommand("command-by-id")
+        , mClusterId(clusterId)
+    {
+        AddArgument("command-id", 0, UINT32_MAX, &mCommandId);
+        AddArgument("payload", &mPayload);
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+        ModelCommand::AddArguments();
+    }
+
+    ClusterCommand(const char * _Nonnull commandName)
+        : ModelCommand(commandName)
+    {
+        AddArgument("timedInteractionTimeoutMs", 0, UINT16_MAX, &mTimedInteractionTimeoutMs);
+        AddArgument("repeat-count", 1, UINT16_MAX, &mRepeatCount);
+        AddArgument("repeat-delay-ms", 0, UINT16_MAX, &mRepeatDelayInMs);
+    }
+
+    ~ClusterCommand() {}
+
+    CHIP_ERROR SendCommand(CHIPDevice * _Nonnull device, chip::EndpointId endpointId) override
+    {
+        chip::TLV::TLVWriter writer;
+        chip::TLV::TLVReader reader;
+        NSDictionary * commandFieldDict = @{@"type" : @"Structure", @"value" : @[]};
+
+        mData = static_cast<uint8_t *>(chip::Platform::MemoryCalloc(sizeof(uint8_t), mDataMaxLen));
+        VerifyOrReturnError(mData != nullptr, CHIP_ERROR_NO_MEMORY);
+
+        writer.Init(mData, mDataMaxLen);
+
+        ReturnErrorOnFailure(mPayload.Encode(writer, chip::TLV::AnonymousTag()));
+        reader.Init(mData, writer.GetLengthWritten());
+        ReturnErrorOnFailure(reader.Next());
+
+        id commandFields = NSObjectFromCHIPTLV(&reader);
+        if (commandFields != nil) {
```

I guess the real question here is whether failure to decode should error out instead of falling back to a default...

---

Yes, that makes sense. I must have not understood. I'll update this and write to error out when a nil is returned. The only thing is the I do not have the error occurred. So I will have to return `CHIP_ERROR_INTERAL`

---

Oook. Updated and tested. I couldn't seem to get the failure scenario to occur though.